TestConcurrencyMultiProcess fails on windows: subprocess build doesn't propagate -tags gms_pure_go

## summary

`TestConcurrencyMultiProcess` in `internal/storage/embeddeddolt/` fails on Windows because the test shells out to `go test -c` to build a subprocess test binary, but doesn't pass `-tags gms_pure_go` through. without that tag the subprocess build pulls in the cgo-backed ICU regex package which needs `unicode/uregex.h` - those headers aren't on a stock Windows Go toolchain so the compile dies before any assertions run.

## repro

windows + go toolchain, no ICU headers installed:

```
BEADS_TEST_EMBEDDED_DOLT=1 BEADS_EMBEDDED_DOLT_PROCS=2 BEADS_EMBEDDED_DOLT_ITERS=1 \
  go test -tags gms_pure_go -count=1 -run "^TestConcurrencyMultiProcess$" -v \
  ./internal/storage/embeddeddolt/...
```

## failure

```
concurrency_test.go:81: build test binary: exit status 1
    # github.com/dolthub/go-icu-regex/internal/icu
    C:\Users\sms\go\pkg\mod\github.com\dolthub\go-icu-regex@v0.0.0-20250916051405-78a38d478790\internal\icu\icu.go:8:11: fatal error: unicode/uregex.h: No such file or directory
        8 | // #include "unicode/uregex.h"
          |           ^~~~~~~~~~~~~~~~~~
    compilation terminated.
--- FAIL: TestConcurrencyMultiProcess (0.91s)
```

## why it doesn't fire on linux/macOS

ICU dev headers are available by default in those toolchains, so the cgo path compiles even without the `gms_pure_go` tag. it's silently fine. Windows is structurally broken without the tag because per CLAUDE.md `gms_pure_go` is mandatory project-wide for builds.

## fix

thread `-tags gms_pure_go` through the `exec.CommandContext("go", "test", "-c", ...)` invocation at `internal/storage/embeddeddolt/concurrency_test.go:73`. PR incoming.

--- a/internal/storage/embeddeddolt/concurrency_test.go
+++ b/internal/storage/embeddeddolt/concurrency_test.go
@@ -69,9 +69,14 @@ func TestConcurrencyMultiProcess(t *testing.T) {
 		t.Logf("using pre-built test binary: %s", testBin)
 	} else {
 		testBin = filepath.Join(t.TempDir(), "embeddeddolt.test")
-		t.Logf("building test binary: go test -c -o %s ./internal/storage/embeddeddolt/", testBin)
+		t.Logf("building test binary: go test -c -tags gms_pure_go -o %s ./internal/storage/embeddeddolt/", testBin)
+		// -tags gms_pure_go is mandatory project-wide (see CLAUDE.md). Without it
+		// the subprocess build pulls in the cgo-backed ICU regex package, which
+		// requires unicode/uregex.h headers that aren't present on Windows
+		// toolchains and breaks the subprocess compile before any assertions run.
 		build := exec.CommandContext(ctx, "go", "test",
 			"-c",
+			"-tags", "gms_pure_go",
 			"-o", testBin,
 			"./internal/storage/embeddeddolt/",
 		)

